### PR TITLE
chore: lower time for fuzz/race tests

### PR DESCRIPTION
**Pull request by erikburt**
Lowers the runtime of CI Core fuzz/race tests during executions triggered by PRs.

### Changes

Race Tests:
- Lower timeouts to 15s, and count to 5 (on PRs only, still at 10m/50 on scheduled runs)
- Add build cache

Fuzz Tests:
- Lower fuzz timeout to 4 minutes on PRs, 10m on scheduled runs
- Calculating `--seconds` parameter passed to python fuzz script, based on FUZZ_TIMEOUT_MINUTES
- Better calculation of time elapsed, and individual fuzzer runtimes in python script

### Testing

https://github.com/smartcontractkit/chainlink/actions/runs/11734083428?pr=15141

- Core Integ Tests: 2m25s
- Deployment Tests: 2m32s
- Race Tests: 4m39s
- Core Tests: 5m34s
- Fuzz Tests: 5m55s

In best case scenario, long pole is now "Clean Go Tidy & Generate"

---

RE-3205

### CODE REVIEW

**Review by chainchad — DISMISSED** [review]

**Review by momentmaker — DISMISSED** [review]

File: tools/bin/go_core_fuzz
Comment (jmank88) [inline tools/bin/go_core_fuzz:1]: Is it OK for these values to be equal? Or was `--seconds` shorter by design to give the script time to exit gracefully?

Comment (jmank88) [reply tools/bin/go_core_fuzz:1]: Oh I misread. Why introduce a new variable? These seem highly related. Can we derive one from the other instead?

Comment (erikburt) [reply tools/bin/go_core_fuzz:1]: c30badb99587080b2dfb442a1298679d38bc4709

**Review by momentmaker — DISMISSED** [review]

**Review by jmank88 — DISMISSED** [review]

**Review by momentmaker — DISMISSED** [review]

**Review by jmank88 — DISMISSED** [review]

**Comment by jmank88** [comment]
Looks good, but could be a little faster? :shrug: 
image

**Comment by erikburt** [comment]
> Looks good, but could be a little faster? 🤷

Ok I've lowered them even further. I was having trouble lowering race tests at all, seems no matter what it sits around 8 minutes runtime. I'm thinking it might be build time that is the majority of the runtime.

**Comment by jmank88** [comment]
> > Looks good, but could be a little faster? 🤷
> 
> Ok I've lowered them even further. I was having trouble lowering race tests at all, seems no matter what it sits around 8 minutes runtime. I'm thinking it might be build time that is the majority of the runtime.

You could try dropping the `-json` flag from the race tests. I don't know why that is enabled, but it creates extra noise by enabling `-v`. Might not matter though :shrug:

**Comment by erikburt** [comment]
> You could try dropping the `-json` flag from the race tests. I don't know why that is enabled, but it creates extra noise by enabling `-v`. Might not matter though 🤷

That seem to have helped by about 30 seconds. Down to 7.5min execution on the race test.

**Comment by erikburt** [comment]
Ok the build cache does help - down to 5m10s...

https://github.com/smartcontractkit/chainlink/actions/runs/11732567357?pr=15141

**Review by jmank88 — APPROVED** [review]

**Review by chainchad — APPROVED** [review]

**Pull request merged**